core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -758,14 +758,12 @@ std::map<int, std::mutex> CudaInternal::constantMemMutexPerDevice     = {};
 
 }  // namespace Kokkos
 
-void Kokkos::Impl::create_Cuda_instances(std::vector<Cuda> &instances) {
-  for (int s = 0; s < int(instances.size()); s++) {
-    cudaStream_t stream;
-    KOKKOS_IMPL_CUDA_SAFE_CALL((
-        instances[s].impl_internal_space_instance()->cuda_stream_create_wrapper(
-            &stream)));
-    instances[s] = Cuda(stream, ManageStream::yes);
-  }
+void Kokkos::Impl::create_Cuda_instance(Cuda &instance) {
+  cudaStream_t stream;
+  KOKKOS_IMPL_CUDA_SAFE_CALL(
+      (instance.impl_internal_space_instance()->cuda_stream_create_wrapper(
+          &stream)));
+  instance = Cuda(stream, ManageStream::yes);
```

Comment:
Similar in HIP, since this is just an impl function, I didn't want to worry about having multiple that could handle array and vector, so I just had it handle a single instance and in CUDA's `partition_space()` we call this function in a loop over the new exec spaces.